src/libasr/location.h

```diff
@@ -152,10 +164,10 @@ struct LocationManager {
     // `in_newlines` starts from 0
     void pos_to_linecol(uint32_t position, uint32_t &line, uint32_t &col) const {
         const std::vector<uint32_t> *newlines;
-        if (preprocessor) {
-            newlines = &in_newlines0;
+        if (files.back().preprocessor) {
+            newlines = &files.back().in_newlines0;
         } else {
-            newlines = &in_newlines;
+            newlines = &files.back().in_newlines;
```

Comment:
In here instead of using the latest file in the `files` stack, it should use `position` and `bisection` in `file_ends` to figure out which file this belongs to. Then it should use the current code to determine `line` and `col` inside the given file.

Reply:
+1 on this approach. This will definitely print the error message with correct information. But, we have to get the index of the files somehow using the Location.

Reply:
I think you implemented it in the below commit, so I think this is resolved.